robot/Cumulus/resources/OpportunityPageObject.py

```diff
@@ -7,7 +8,68 @@
 class OpportunityPage(BaseNPSPPage, DetailPage):
     object_name = "Opportunity"
 
-    
+    def _is_current_page(self):
+        """ Verify we are on the opportunity details page
+            by verifying that the url contains '/view'
+        """
+        self.selenium.location_should_contain("/lightning/r/Opportunity/",message="Current page is not a Opportunity detail view")
+
+
+    def verify_payments_made(self, count):
+        """
+            Verify payments tally matches the count specified on the opportunity details page
+        """
+
+        self.npsp.select_tab("Related")
+        self.salesforce.load_related_list("Payments")
+        self.npsp.verify_occurrence("Payments", count)
+
+
+    def validate_values_under_relatedlist_for(self, listname, **kwargs):
+        """
```

Comment:
these two are common keywords that can be used for any related list, not just payments. Also instead of creating seperate keyword, I would suggest to update 'verify occurrence' keyword to do the same thing as 'verify payments made' keyword. Same thing for 'validate values under relatedlist for' keyword move everything to verify related list field values. Please feel free to change names if u want.

Reply:
Addressed this .